src/clusterfuzz/_internal/build_management/build_manager.py

```diff
@@ -402,7 +404,70 @@ def _post_setup_success(self, update_revision=True):
     if instrumented_library_paths:
       self._patch_rpaths(instrumented_library_paths)
 
-  def _unpack_build(self, base_build_dir, build_dir, build_url):
+  @contextlib.contextmanager
+  def _open_build_archive(self, base_build_dir: str, build_dir: str,
+                          build_url: str, http_build_url: Optional[str]):
+    """Gets a handle on a build archive for the current build. Depending on the
+    provided parameters, this function might download the build archive into
+    the build directory or directly use remote HTTP archive.
+
+    Args:
+        base_build_dir: the base build directory.
+        build_dir: the current build directory.
+        build_url: the build URL.
+        http_build_url: the HTTP build URL. Defaults to None.
+
+    Raises:
+        if an error occurred while accessing the file over HTTP or while
+        downloading the file on disk.
+
+    Returns:
+        the build archive.
+    """
+    unpack_everything = environment.get_value(
+        'UNPACK_ALL_FUZZ_TARGETS_AND_FILES')
+
+    # We only want to use remote unzipping if we're not unpacking everything and
+    # if the HTTP URL is compatible with remote unzipping.
+    can_unzip_over_http = not unpack_everything and http_build_url\
+      and build_archive.unzip_over_http_compatible(http_build_url)
+
+    if not can_unzip_over_http:
+      # Download build archive locally.
+      build_local_archive = os.path.join(build_dir, os.path.basename(build_url))
+
+      # Make the disk space necessary for the archive available.
+      archive_size = storage.get_object_size(build_url)
+      if archive_size is not None and not _make_space(archive_size,
+                                                      base_build_dir):
+        shell.clear_data_directories()
+        logs.log_fatal_and_exit(
+            'Failed to make space for download. '
+            'Cleared all data directories to free up space, exiting.')
+
+      logs.info(f'Downloading build from {build_url}.')
+      try:
+        storage.copy_file_from(build_url, build_local_archive)
+      except Exception as e:
+        logs.error(f'Unable to download build from {build_url}: {e}')
+        raise
+
+      try:
+        with build_archive.open(build_local_archive) as build:
+          yield build
```

Comment:
No, I kind of need the context manager here because I want to delete the file after we used it. I changed the code a bit, and I think that's cleaner, WDYT?